File: back-end/src/main/java/com/action/headline/controller/NewsTypeController.java
package com.action.headline.controller;


import com.action.headline.common.Result;
import com.action.headline.entity.NewsType;
import com.action.headline.service.NewsTypeService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/types")
public class NewsTypeController {

    @Autowired
    private NewsTypeService typeService;

    /**
     * 获取所有新闻类型
     * @return
     */
    @GetMapping
    public Result findAll() {
        List<NewsType> newsTypeList = typeService.findAll();
        return Result.ok(newsTypeList);
    }
}

File: back-end-gin/controller/news_type_controller.go
package controller

import (
	"github.com/action/micro-headlines/common"
	"github.com/action/micro-headlines/service"
	"github.com/gin-gonic/gin"
)

type NewsTypeController struct {
	typeService *service.NewsTypeService
}

func NewNewsTypeController(typeService *service.NewsTypeService) *NewsTypeController {
	return &NewsTypeController{typeService: typeService}
}

// FindAll 获取所有新闻类型
func (ctrl *NewsTypeController) FindAll(c *gin.Context) {
	types, err := ctrl.typeService.FindAll()
	if err != nil {
		common.Error(c, common.SYSTEM_UNKNOWN_ERROR, err.Error())
		return
	}
	
	common.Ok(c, types)
}

